its/tests/src/test/java/its/SonarQubeDeveloperEditionTests.java

```diff
@@ -302,7 +340,11 @@ void shouldRaiseIssuesOnAPhpProject() throws Exception {
       provisionProject(ORCHESTRATOR, projectKey, "Sample PHP");
       ORCHESTRATOR.getServer().restoreProfile(FileLocation.ofClasspath("/php-sonarlint.xml"));
       ORCHESTRATOR.getServer().associateProjectToQualityProfile(projectKey, "php", "SonarLint IT PHP");
-      updateProject(engine, projectKey);
+
+      backend.getConfigurationService().didAddConfigurationScopes(new DidAddConfigurationScopesParams(
+        List.of(new ConfigurationScopeDto(CONFIG_SCOPE_ID, null, true, "projectName", new BindingConfigurationDto(CONNECTION_ID, projectKey, true)))));
+      await().atMost(20, SECONDS).untilAsserted(() -> assertThat(didSynchronizeConfigurationScopes).contains(CONFIG_SCOPE_ID));
+      await().untilAsserted(() -> assertThat(clientLogs.stream().anyMatch(s -> s.getMessage().equals("Stored project analyzer configuration"))).isTrue());
```

Comment:
The fact that you need to await for logs is a red flag to me. We don't want real clients to do the same. Isn't the DidSynchronizeConfigurationScope enough?

Reply:
Checking for the `"Stored project analyzer configuration"` was added mostly for AnalysisTests. Without it, the engine.analyze() throws  (on cirrus, not reproducible locally) `java.nio.file.NoSuchFileException: ...../analyzer_config.pb` as it's still being updated (or at least the lock isn't released) even though we receive the DidSynchronizeConfigurationScope event.
I agree that waiting for a log is bad. I haven't had a better idea.